bitcoin/src/psbt/serialize.rs

```diff
@@ -173,6 +174,32 @@ impl Deserialize for secp256k1::PublicKey {
     }
 }
 
+impl Serialize for Vec<secp256k1::PublicKey> {
+    fn serialize(&self) -> Vec<u8> {
+        let mut result: Vec<u8> = Vec::with_capacity(secp256k1::constants::PUBLIC_KEY_SIZE * self.len());
+
+        for pubkey in self.iter() {
+            result.extend(Serialize::serialize(pubkey));
+        }
+
+        result
+    }
+}
+
+impl Deserialize for Vec<secp256k1::PublicKey> {
+    fn deserialize(bytes: &[u8]) -> Result<Self, Error> {
+        bytes.chunks(secp256k1::constants::PUBLIC_KEY_SIZE)
+            .map(|pubkey_bytes| {
+                if pubkey_bytes.len() == secp256k1::constants::PUBLIC_KEY_SIZE {
+                    secp256k1::PublicKey::deserialize(pubkey_bytes)
+                } else {
+                    Err(Error::ConsensusDeserialize(DeserializeError::Parse(ParseError::MissingData)))
```

Comment:
Is there a better choice of error variant?

Reply:
I'd probably use the same error as `Deserialize` for `PublicKey` uses. So this can be simplified to 

```rust
        bytes.chunks(secp256k1::constants::PUBLIC_KEY_SIZE)
            .map(|pubkey_bytes| {
                secp256k1::PublicKey::deserialize(pubkey_bytes)
            })
            .collect()
```